notion-server/src/ethereum.rs

```diff
@@ -14,24 +87,40 @@ impl EthereumSimulation {
         }
     }
 
+    pub fn get_general_simulation_state(&self) -> interface_args::GeneralSimulationState {
```

Comment:
Getters in Rust generally don't include the `get_` prefix. Perhaps `pub fn simulation_state(&self)`?

Reply:
Works for me!  I always like to learn more about Rust packages and Rust conventions / tips, thanks for including all of them and please keep them coming!